### PR TITLE
Changes to show the right amount of time

### PR DESCRIPTION
**Pull request by JoaquinJimenezGarcia**
Hi @jeevatkm !

We have been working in your project and did some little modifications due we didn't have the right time like said in #304  ("TotalTime" wasn't showing the sum of the rest of points, because "ConnTime" wasn't been calculating with its right values).

For that, we added ConnectStart and ConnectDone and also changed the way ConnTime was been calculating, and now we have the right time as shown below (request to https://www.google.es).

times_google es

### CODE REVIEW

File: request.go [hunk 1]
```diff
@@ -564,10 +564,13 @@ func (r *Request) TraceInfo() TraceInfo {
 	if ct == nil {
 		return TraceInfo{}
 	}
+	if ct.dnsStart.IsZero() {
```
Comment (moorereason) [inline request.go:567]: Why move this out of `clientTrace.createContext`?  It feels out of place here.  Can you put this in the `ConnectStart` hook where you have the `dnsDone` check?

Comment (JoaquinJimenezGarcia) [reply request.go:567]: OK, I change it!

File: request.go [hunk 2]
```diff
@@ -564,10 +564,13 @@ func (r *Request) TraceInfo() TraceInfo {
...
+		ct.dnsStart = ct.dnsDone
+	}
 
 	return TraceInfo{
 		DNSLookup:     ct.dnsDone.Sub(ct.dnsStart),
-		ConnTime:      ct.gotConn.Sub(ct.getConn),
+		ConnTime:      ct.connectDone.Sub(ct.dnsDone),
```
[the remark this answers is not in the document]
Comment (JoaquinJimenezGarcia) [reply request.go:573]: OK, we need another extra time to know TCP Connection. So we mantain your time and we will create a new one called TCPConnTime as done in the following url:

- https://github.com/davecheney/httpstat (referenced on: https://blog.golang.org/http-tracing)

File: trace.go [hunk 3]
```diff
@@ -27,6 +27,8 @@ type TraceInfo struct {
 	// ConnTime is a duration that took to obtain a successful connection.
 	ConnTime time.Duration
 
+	TCPConnTime time.Duration
```
Comment (moorereason) [inline trace.go:30]: This needs a godoc comment.

Comment (JoaquinJimenezGarcia) [reply trace.go:31]: Added!

Comment (moorereason) [reply trace.go:31]: Needs an ending period.  😃 Best practice is to use complete sentences for godoc comments.  Also, capitalize TCP.

Comment (JoaquinJimenezGarcia) [reply trace.go:31]: Thank you for your comment @moorereason !

Added in this new commit!

**Comment by moorereason** [comment]
Thanks, @JoaquinJimenezGarcia.

@jeevatkm, I'm happy with this PR, so I'll leave it to you for further review.

**Comment by toni-moreno** [comment]
Hello @moorereason @jeevatkm. I'm trying to test this release but  I can't import in my gomodules if the branch is not under the correct name "github.com/go-resty/resty" istead of any other "github.com/xxxxxxx/resty".  Could you please merge or at least create a new branch for this PR ? . Thank you very much.

**Comment by moorereason** [comment]
@toni-moreno,
First, I'm just a Resty user. :smiley:  Second, I checkout PRs with the following bash function (i.e. `prco 331`):

```bash
function prco() {
        if [ $# -lt 1 ]; then
                echo "error: missing pull request ID"
                return
        fi

        git fetch origin refs/pull/$1/head:PR-$1 && \
        git checkout PR-$1
}
```

**Comment by toni-moreno** [comment]
Hi @moorereason , I'm trying to compile my tool with this patch, importing it with go modules (.

And this happens

````bash
$ go get -u github.com/JoaquinJimenezGarcia/resty/v2@issue/correct_amount_of_time
go: finding github.com issue/correct_amount_of_time
go: finding github.com/JoaquinJimenezGarcia/resty/v2 issue/correct_amount_of_time
go: finding github.com/JoaquinJimenezGarcia/resty issue/correct_amount_of_time
go: finding github.com/JoaquinJimenezGarcia issue/correct_amount_of_time
go get github.com/JoaquinJimenezGarcia/resty/v2@issue/correct_amount_of_time: github.com/JoaquinJimenezGarcia/resty/v2@issue/correct_amount_of_time: invalid version: version "issue/correct_amount_of_time" invalid: disallowed version string
````

I've forked and renamed the branch into `https://github.com/datadope-io/resty/tree/fix_http_trace_time` trying to fix this error . And another error happen

````bash
$ go get -u github.com/datadope-io/resty/v2@fix_http_trace_time
go: finding github.com/datadope-io/resty fix_http_trace_time
go: finding github.com fix_http_trace_time
go: finding github.com/datadope-io fix_http_trace_time
go: finding github.com/datadope-io/resty/v2 fix_http_trace_time
go get: github.com/datadope-io/resty/v2@v2.2.1-0.20200427074555-c73dc22cb82d: parsing go.mod:
	module declares its path as: github.com/go-resty/resty/v2
	        but was required as: github.com/datadope-io/resty/v2
````

This is the reason because of I ask for create inside github.com/go-resty/resty branch while waiting to merge.

**Comment by moorereason** [comment]
Learn to use the “replace” directive in your go.mod file to point to a local path.

**Comment by toni-moreno** [comment]
@moorereason  Thank you for your suggestion.
My  first idea was do it without code changes  and work direct from the mainstream repo, I would like  do that by only import a branch `go get -u github.com/go-resty/resty/v2@branch`

**Pull request merged**